apps/wpcom-block-editor/src/wpcom/features/use-site-intent.js

```diff
@@ -0,0 +1,28 @@
+import apiFetch from '@wordpress/api-fetch';
+import { useState, useEffect, useCallback } from '@wordpress/element';
+
+// FIXME: We can use `useSiteIntent` from `@automattic/data-stores` and remove this.
+// https://github.com/Automattic/wp-calypso/pull/73565#discussion_r1113839120
```

Comment:
Using @automattic/data-stores requires TypeScript support, which is its own can of worms, so I opted to reuse this method to try and get a proof of concept working.